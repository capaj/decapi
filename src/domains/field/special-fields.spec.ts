import { ObjectType, Query, compileObjectType } from '../..'

describe('Special fields - @Query, @Mutation @Subscribe', () => {
  it('Will not allow registering special type on type that is not @SchemaRoot', () => {
    @ObjectType()
    class Foo {
      @Query()
      bar(): string {
        return null
      }
    }

    expect(() => {
      return compileObjectType(Foo).getFields()
    }).toThrowErrorMatchingInlineSnapshot(
      `"@ObjectType Foo.bar: Given field is root field (@Query or @Mutation) not registered inside @SchemaRoot type. "`
    )
  })
})
