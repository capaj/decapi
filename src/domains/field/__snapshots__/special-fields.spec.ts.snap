// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Special fields - @Query, @Mutation @Subscribe Will not allow registering special type on type that is not @Schema 1`] = `"@ObjectType Foo.bar: Given field is root field (@Query or @Mutation) not registered inside @Schema type. "`;
